src/gui/qgsprovidersourcewidget.cpp

```diff
@@ -0,0 +1,25 @@
+/***************************************************************************
+                             qgsprovidersourcewidget.cpp
+                             -----------------
+    begin                : August 2018
```

Comment:
Back to the future.

Reply:
Believe it or not, that's actually when I started this work...